### DISCUSSION TITLE
Cant select item after I use it

### DISCUSSION THREAD

**Discussion by doodleswodles** [peter-kish/gloot#138] [General]
Here's my code it is in an inventory script in the control node at the root of the scene

```
 func _on_ctrl_inventory_grid_ex_inventory_item_activated(item):
	var itemName = item.get_property("id")
	var currentItemEffect 
	match itemName :
		"needle": 
			currentItemEffect = GlobalVar.items.needle.effect
		"bandage": 
			currentItemEffect = GlobalVar.items.bandage.effect
	currentItemEffect.call()
	GlobalVar.set_health_bar(GlobalVar.player.Adrey.hp)
	var stack_size: int = item.get_property("stack_size")
	InventoryGridStacked.set_item_stack_size(item, stack_size - 1)  
```

**Comment by doodleswodles** [comment]
the problem is in the deleting code
but Idk how to fix it

**Reply by peter-kish** [reply]
Can you give a bit more info on this? I'm not sure what exactly the problems is here. When you say "deleting code", I suppose you're referring to this part:
```gdscript
var stack_size: int = item.get_property("stack_size")
InventoryGridStacked.set_item_stack_size(item, stack_size - 1)
```
where the item will be removed and queued for deletion when the stack size reaches 0.
But I don't see any issues here just by looking at it...

**Reply by doodleswodles** [reply]
I fixed it by making it a function called in physics process with an input instead of the function that came with the addon because it is bugged
